core/src/main/java/org/wso2/carbon/kernel/config/model/CarbonConfiguration.java

```diff
@@ -16,29 +16,39 @@
 package org.wso2.carbon.kernel.config.model;
 
 import org.wso2.carbon.kernel.Constants;
+import org.wso2.carbon.kernel.annotations.Configuration;
+import org.wso2.carbon.kernel.annotations.Element;
 import org.wso2.carbon.kernel.internal.config.JMXConfiguration;
 
 /**
- * CarbonConfiguration class holds static configuration parameters specified in the carbon.yaml file.
+ * CarbonConfiguration class holds static configuration parameters.
  *
  * @since 5.0.0
  */
+@Configuration(namespace = "wso2.carbon", description = "Carbon Configuration Parameters")
 public class CarbonConfiguration {
 
+    @Element(description = "value to uniquely identify a server", required = true)
     private String id = "carbon-kernel";
 
+    @Element(description = "server name")
     private String name = "WSO2 Carbon Kernel";
 
-    private String version = "5.0.0";
+    @Element(description = "server version")
+    private String version = "5.2.0-SNAPSHOT";
```

Comment:
Would it be possible to set the version value at the build time by reading the version in pom.xml file? Otherwise, when this component being released, this value might need to be manually updated.